compio-driver/src/poll/mod.rs

```diff
@@ -176,41 +171,28 @@ impl Driver {
         })
     }
 
+    pub fn create_op<T: crate::sys::OpCode + 'static>(&self, user_data: usize, op: T) -> RawOp {
+        RawOp::new(user_data, op)
+    }
+
     fn submit(&mut self, user_data: usize, arg: WaitArg) -> io::Result<()> {
```

Comment:
The RawFd passed in is possibly invalid (when only considering this function). We either need to mark `submit` as `unsafe`, or by adding the validity constraint to `WaitArg` (by providing faillible/unsafe constructors).

Reply:
I think it's safe. Either epoll or kqueue will return error of "invalid fd".

Reply:
> I think it's safe. Either epoll or kqueue will return error of "invalid fd".

That's guarantee from outside. Locally this is unsafe.

Reply:
OK.